scss/typography/_dropcaps.scss

```diff
@@ -0,0 +1,63 @@
+////
+/// @group typography-dropcaps
+////
+
+/// Default number of lines against the rest of the content  
+/// @type Number
+$dropcaps-lines: 3;
+
+/// Default line height for the rest of the content
+/// @type Number
+$dropcaps-line-height: 1.6;
+
+/// If `true`, makes dropcaps uppercase.
+/// @type Boolean
+$dropcaps-uppercase: true !default;
+
+/// Dropcaps component
+/// @param { Number } $lines [$dropcaps-lines] number of lines against the rest of the content 
+/// @param { Number } $line-height [$dropcaps-line-height] Default line height for the rest of the content
+/// @param { Boolean } $uppercase [$dropcaps-uppercase] If `true`, makes dropcaps uppercase.
+@mixin dropcaps(
+  $lines: $dropcaps-lines,
+  $line-height: $dropcaps-line-height,
+  $uppercase: $dropcaps-uppercase
+) {
+  // Default margin right (Arial/Helvetica) for having a tiny space between them through the right
+  // See: http://code.stephenmorley.org/html-and-css/creating-drop-caps/
+  $-margin-right: 0.0962em;
+  $mt: (15 / 100); // Distance between Ascent and Top of the font
+  $mb: (18 / 100); // Distance between Baseline and Bottom of the font
+
+  $lh: $line-height; // Defining $line-height as `$lh` for easy calculations
+  $n: $lines; // Defining $lines as `$n` for easy calculations
+
+  $h: $n * $lh; // Multiplied line height and number of lines to a variable `$h`
```

Comment:
I think math variables do not need any comment. It is clear enough it is aliases for easier calculations.